public/js/results.js

```diff
@@ -2,74 +2,78 @@
+	}
 
-// actions (do stuff)
-View.setHeader("Results", ElectionId)
-View.statusbar.innerHTML = ""
-Piv.electionsMenu(View.sidenav, ElectionId)
+	var group = Piv.div(ResultsList, "", "border-color-3 bg-color-white")
+	for (var key in candidates) {
+	    var name = candidates[key].name
+	    group.innerHTML += ("<b>Rank " + rank + ":</b> " + name + "<br>")
+	}
+    }
 
-Piv.removeHrefsForCurrentLoc()  //remove hrefs that link to the current page
+    function getLastResultSnapshot(snapshots) {
+	if (snapshots.length < 1) {
+	    showErrorMessage()
+	    return
+	}
+	Piv.http.get(["/api/elections/" + ElectionId + "/result_snapshots/" + snapshots[snapshots.length - 1].id], showElectionResults, showErrorMessage)
+    }
 
-// Piv.http.get(["/api/elections/" + ElectionId + "/result"], showElectionResults, showErrorMessage)
-Piv.http.get(["/api/elections/" + ElectionId + "/result_snapshots"], getLastResultSnapshot, showErrorMessage)
-// Piv.http.get(["/api/elections/" + ElectionId + "/result_snapshots"])
-// Piv.http.post(["/api/elections/" + ElectionId + "/result_snapshots"])
-// Piv.http.get(["/api/elections/" + ElectionId + "/result"])
+    function showElectionResults(results) {
+	console.log(results)
+	var version = parseInt(results.format_version) // TODO: determine why this is a string in some deployments
+	var supported = [VERSION_SURFACE_CANDIDATE_TIES]
+	if (supported.indexOf(version) >= 0) {
+	    if ('error' in results.result_blob && results.result_blob['error'] != null) {
+		Piv.div(View.workspace, "", "100 text3", results.result_blob['error'])
+		return
+	    }
 
-// function definitions
-function displayCandidate(parent, description) {
-  // var candidateLiAtts = {"class": "w100 border-bottom-2 overflow-visible hover-1 drag-drop-1", "data-id": id}
-  var candidateLiAtts = {"class": "w100 border-bottom-1 overflow-visible"}
-  var box = Piv.html(parent, "li", "", candidateLiAtts);
-  Piv.div(box, "", "text1square orderdisplay");
-  Piv.div(box, "", "text3 w75", description)
-  // Piv.div(box, "", "text1 w75", description);
-}
-function getLastResultSnapshot(snapshots) {
-  if (snapshots.length < 1) {
-    showErrorMessage()
-    return
-  }
-  Piv.http.get(["/api/elections/" + ElectionId + "/result_snapshots/" + snapshots[snapshots.length - 1].id], showElectionResults, showErrorMessage)
-}
-function showElectionResults(results) {
-  console.log(results)
-  var version = parseInt(results.format_version) // TODO: determine why this is a string in some deployments
-  var supported = [VERSION_ADD_RESULTS, VERSION_ADD_DEBUG, VERSION_ADD_ERROR_INFO, VERSION_ADD_ELECTOR_INFO]
-  if (supported.indexOf(version) >= 0) {
-    if ('error' in results.result_blob && results.result_blob['error'] != null) {
-      Piv.div(View.workspace, "", "100 text3", results.result_blob['error'])
-      return
+	    var candidateOrder = results.result_blob.order
+	    ResultsList.innerHTML = ""
+	    var rank = 1
+	    for (var key in candidateOrder) {
```

Comment:
So feel free to continue using for in loops. It might still be handy to read the blog post so that you are aware of this gradually-fading pitfall and so that you can avoid libraries that add enumerable properties to built-in object prototypes.

Reply:
Yuck, I would have expected "for (var x in values)" to be syntactic sugar for "values.forEach(function(x){...})".  Good to know that's not the case.  I'll try to replace the former with the latter (in one or two other places as well).

Reply:
Yeah, unfortunately it's not an exact replacement. As time goes on more library developers will start making their prototype properties non-enumerable, so it's going to get better and maybe someday be good enough to use `for in` safely. It's just a question of when does the asymptote get "close enough" to the limit?